ffmpeg/README.md

```diff
@@ -0,0 +1,119 @@
+    autoconf \
+    automake \
+    build-essential \
+    cmake \
+    git \
+    libtool \
+    pkg-config \
+    texinfo \
+    wget \
+    yasm \
+    zlib1g-dev
+    ```
+1. On the virtual machine, clone and cd into the [ffmpeg GitHub repository](https://github.com/ffmpeg/ffmpeg) by running
+
+    ```
+    git clone https://github.com/ffmpeg/ffmpeg && \
+    cd ffmpeg
+    ```
+1. Checkout the release you want to compile by running
+
+    ```
+    git fetch --all --tags --prune && \
+    git checkout tags/<tag_name> -b <branch_name>
+    ```
+    where `<tag_name>` is the name of the tag you want to checkout, and `<branch_name>` is the name of a new local branch to checkout that commit on.
+1. On the virtual machine, configure the source for compilation & make by running
+
+    ```
+    ./configure \
+    --disable-shared \
+    --enable-static \
+    --pkg-config-flags="--static" \
+    --extra-cflags="-static" \
+    --extra-ldflags="-static" \
+    --enable-version3 && \
+    make
+    ```
+1. The root directory of the ffmpeg repo will now have the desired binary `ffmpeg`.
+
+### Windows:
+
+1. [Install Vagrant](https://www.vagrantup.com/downloads.html).
+1. Clone and cd into this [Windows Vagrant GitHub repository](https://github.com/oconnormi/vagrant-windows) made by Michael O'Connor by running
+
+    ```
+    git clone https://github.com/oconnormi/vagrant-windows
+    ```
+1. Run `vagrant up`.
+1. Run `vagrant rdp`.
+1. On the virtual machine, [download MSYS2 32 bit](http://repo.msys2.org/distrib/i686/msys2-i686-20161025.exe).
+1. On the virtual machine, start `mingw32.exe` found under `C:\msys32`. This will be our terminal for the remainder of this guide.
+1. On the virtual machine, install compilation dependencies by running
+
+    ```
+    pacman -S make \
+    diffutils \
+    yasm \
+    mingw-w64-i686-gcc
+    ```
+1. On the virtual machine, clone and cd into the [ffmpeg GitHub repository](https://github.com/ffmpeg/ffmpeg) by running
+
+    ```
+    git clone https://github.com/ffmpeg/ffmpeg && \
+    cd ffmpeg
+    ```
+Note that mingw32 doesn't have git installed. You can open `Git Bash` to use git, but make sure to switch back to mingw32 for the remainder of this guide.
+1. Checkout the release you want to compile by running
+
+    ```
+    git fetch --all --tags --prune && \
+    git checkout tags/<tag_name> -b <branch_name>
+    ```
+    where `<tag_name>` is the name of the tag you want to checkout, and `<branch_name>` is the name of a new local branch to checkout that commit on.
+1. On the virtual machine, configure the source & make by running
+
+    ```
+    ./configure \
+    --disable-shared \
+    --enable-static \
+    --pkg-config-flags="--static" \
+    --extra-cflags="-static" \
+    --extra-ldflags="-static" \
+    --enable-version3 && \
+    make
+    ```
+1. The root directory of the cloned ffmpeg repo will now have the desired binary `ffmpeg.exe`.
+
+### Testing
```

Comment:
Here's the section for OSX instructions that I followed:
```
### OS X
> Note
>
> FFMPEG was compiled directly on an OS X system rather than a VM because it was simpler.

1. Install compilation dependencies by running
    ```
    brew install automake fdk-aac git lame libass libtool libvorbis libvpx \
    opus sdl shtool texi2html theora wget x264 x265 xvid nasm
    ```
1. Configure the source and make by running
    ```
    git clone http://source.ffmpeg.org/git/ffmpeg.git ffmpeg
    cd ffmpeg
    ./configure --extra-cflags='-arch i386' \
    --extra-ldflags='-arch i386' \
    --arch=x86_32 \
    --target-os=darwin \
    --enable-cross-compile \
    --pkg-config-flags=--static \
    --enable-static \
    --disable-shared \
    --enable-version3
    make
```